keps/sig-node/2033-rootless/README.md

```diff
@@ -0,0 +1,797 @@
+cause increased failures in production. See more in the PRR KEP at
+https://git.k8s.io/enhancements/keps/sig-architecture/20190731-production-readiness-review-process.md.
+
+The production readiness review questionnaire must be completed for features in
+v1.19 or later, but is non-blocking at this time. That is, approval is not
+required in order to be in the release.
+
+In some cases, the questions below should also have answers in `kep.yaml`. This
+is to enable automation to verify the presence of the review, and to reduce review
+burden and latency.
+
+The KEP must have a approver from the
+[`prod-readiness-approvers`](http://git.k8s.io/enhancements/OWNERS_ALIASES)
+team. Please reach out on the
+[#prod-readiness](https://kubernetes.slack.com/archives/CPNHUMN74) channel if
+you need any help or guidance.
+
+-->
+
+### Feature Enablement and Rollback
+
+_This section must be completed when targeting alpha to a release._
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [ ] Feature gate (also fill in values in `kep.yaml`)
+  - [X] Other
+    - Describe the mechanism: By running kubelet and kube-proxy in a user namespace created by RootlessKit.
+
+* **Does enabling the feature change any default behavior?**
+
+No.
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+
+Yes, by just switching away to traditional rootful setup.
+
+* **What happens if we reenable the feature if it was previously rolled back?**
+
+It works again.
+
+* **Are there any tests for feature enablement/disablement?**
+
+Yes, see [Usernetes smoke tests](https://github.com/rootless-containers/usernetes/blob/v20201118.0/.github/workflows/main.yaml).
+
+### Rollout, Upgrade and Rollback Planning
+
+_This section must be completed when targeting beta graduation to a release._
+
+This KEP does not modify Pod specs, so there is no new topic about rolling out, upgrading, or rolling back pod replicas.
+
+### Monitoring Requirements
+
+_This section must be completed when targeting beta graduation to a release._
+
+* **How can an operator determine if the feature is in use by workloads?**
+
+SSH into a node, and see whether kubelet is running as a non-root user.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+  - [ ] Metrics
+    - Metric name:
+    - [Optional] Aggregation method:
+    - Components exposing the metric:
+  - [X] Other (treat as last resort)
+    - Details: Use `systemctl --user is-system-running` to verify whether the processes (RootlessKit, kubelet, kube-proxy, and CRI) are running.
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+
+99% of `systemctl --user is-system-running` per day finish without an error.
+
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?**
+
+No.
```

Comment:
Not required for alpha, but I'll mention it now anyway: it'd be useful to have the kubelet publish whether or not it is running rootless, as a boolean metric.

Reply:
updated
```md

* **Are there any missing metrics that would be useful to have to improve observability 
of this feature?**

N/A, but it'd be useful to have the kubelet publish whether or not it is running rootless, as a boolean metric.
```